src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -108,24 +107,23 @@ float64_t CGaussianKernel::compute(int32_t idx_a, int32_t idx_b)
 {
 	if (!m_compact)
 	{
-		float64_t result=sq_lhs[idx_a]+sq_rhs[idx_b]
-				-2*CDotKernel::compute(idx_a, idx_b);
-		return CMath::exp(-result/width);
+		float64_t result=distance(idx_a,idx_b);
+		return CMath::exp(-result);
 	}
 
 	int32_t len_features, power;
-	len_features=((CDenseFeatures<float64_t>*) lhs)->get_num_features();
+	len_features=((CDotFeatures*) lhs)->get_dim_feature_space();
 	power=(len_features%2==0) ? (len_features+1):len_features;
 
-	float64_t result=sq_lhs[idx_a]+sq_rhs[idx_b]-2*CDotKernel::compute(idx_a,idx_b);
-	float64_t result_multiplier=1-(sqrt(result/width))/3;
+	float64_t result=distance(idx_a,idx_b);
+	float64_t result_multiplier=1-(sqrt(result))/3;
 
 	if (result_multiplier<=0)
 		result_multiplier=0;
 	else
 		result_multiplier=pow(result_multiplier, power);
 
-	return result_multiplier*exp(-result/width);
+	return result_multiplier*exp(-result);
```

Comment:
Please add a unit test for the Gaussina kernel against some matlab or python script with a length scale different to 1, just to make sure results dont change